doc/Security.xml

```diff
@@ -947,13 +991,311 @@ Added certificate-based client authentication</revremark>
+                </entry>
+                <entry>
+                  <para>Authorization server address</para>
+                </entry>
+              </row>
+              <row>
+                <entry>
+                  <para>ClientId</para>
+                </entry>
+                <entry>
+                  <para>Client identifier issued by the authorization server</para>
+                </entry>
+              </row>
+              <row>
+                <entry>
+                  <para>ClientSecret</para>
+                </entry>
+                <entry>
+                  <para>Client secret used to authenticate with the authorization server</para>
+                </entry>
+              </row>
+              <row>
+                <entry>
+                  <para>Scope</para>
+                </entry>
+                <entry>
+                  <para>The requested access scope(s)</para>
+                </entry>
+              </row>
+              <row>
+                <entry>
+                  <para>Type</para>
+                </entry>
+                <entry>
+                  <para>The type of configuration, see <xref xmlns:xlink="http://www.w3.org/1999/xlink" linkend="table_ry3_fbr_swb"/> for
+                    possible values.</para>
+                </entry>
+              </row>
+              <row>
+                <entry>
+                  <para>ClientAuth</para>
+                </entry>
+                <entry>
+                  <para>The type of client authentication method to use, see <xref xmlns:xlink="http://www.w3.org/1999/xlink" linkend="table_smg_rzq_swb"/> for
+                    possible values.</para>
+                </entry>
+              </row>
+            </tbody>
+          </tgroup>
+        </table>
+        <para>The following types of authorization server configurations are defined.</para>
+        <table xml:id="table_ry3_fbr_swb">
+          <title>Authorization server configuration types</title>
+          <tgroup cols="2">
+            <colspec colname="c1" colwidth="30*"/>
+            <colspec colname="c2" colwidth="70*"/>
+            <thead>
+              <row>
+                <entry>
+                  <para>Method</para>
+                </entry>
+                <entry>Description</entry>
+              </row>
+            </thead>
+            <tbody valign="top">
+              <row>
+                <entry>
+                  <para>OAuthAuthorizationCode</para>
+                </entry>
+                <entry>
+                  <para>OAuth2 authorization code flow per [RFC 6749].</para>
+                </entry>
+              </row>
+              <row>
+                <entry>
+                  <para>OAuthClientCredentials</para>
+                </entry>
+                <entry>
+                  <para>OAuth2 client credentials grant flow per [RFC 6749].</para>
+                </entry>
+              </row>
+              <row>
```

Comment:
This enum value is only documented in the spec, not the wsdl (missing?). Also, if we have a distinction between OAuth and OIDC, doesn't that mean that we should also have OIDC2ClientCredentials, assuming that OIDC2 entries read the .well-known/openid-configuration path and the OAuth entries expect the token URI directly.

I think we should update the spec to indicate what is the expected ServerUri in each scenario (authority, discovery document, authorization endpoint and token endpoint)